x/storage/keeper/cross_app_bucket.go

```diff
@@ -204,7 +224,24 @@ func (app *BucketApp) handleCreateBucketFailAckPackage(ctx sdk.Context, appCtx *
 	return sdk.ExecuteResult{}
 }
 
+func (app *BucketApp) handleCreateBucketFailAckPackageV2(ctx sdk.Context, appCtx *sdk.CrossChainAppContext, synPackage *types.CreateBucketSynPackageV2) sdk.ExecuteResult {
+	app.storageKeeper.Logger(ctx).Error("received create bucket fail ack package ")
+
+	return sdk.ExecuteResult{}
+}
+
 func (app *BucketApp) handleCreateBucketSynPackage(ctx sdk.Context, appCtx *sdk.CrossChainAppContext, createBucketPackage *types.CreateBucketSynPackage) sdk.ExecuteResult {
+	if ctx.IsUpgraded(upgradetypes.Pampas) {
+		return sdk.ExecuteResult{
```

Comment:
Do we need `err := createBucketPackage.ValidateBasic()` ?

Reply:
It's on the below